### ISSUE TITLE
deps: provider should avoid importing `github.com/hashicorp/vault`

### ISSUE THREAD

**Issue by jrasell** [hashicorp/terraform-provider-nomad#150]
Vault issue comment reference: https://github.com/hashicorp/vault/issues/9575#issuecomment-665013482

The Nomad provider currently imports `github.com/hashicorp/vault` because of its use of `config.DefaultTokenHelper()` within `provider.go`. A per the above discussion Vault imports should be limited to `sdk` and `api` otherwise problems may occur.

A quick look indicates there is no token helper functionality within either of the recommended packages, although this was not exhaustive.

**Comment by tristanmorgan** [comment]
A note for anyone caught when you broke your Vault token helper.... this is the error message displayed.
```
╷
│ Error: Error getting token helper: unknown error getting the external helper path
│ 
│   with provider["registry.terraform.io/hashicorp/nomad"],
│   on <input-prompt> line 1:
│   (source code not available)
│ 
╵
```
Is there a reason why a vault misconfiguration should affect a 3 line nomad config?
```hcl
data "nomad_job" "example" {
  job_id = "example"
}
```

**Comment by lgfa29** [comment]
Hi @tristanmorgan,

The Nomad provider uses the same logic as Vault to find a token, so if your Vault token helper is incorrect, the Nomad provider will fail as well.

**Comment by lgfa29** [comment]
Closed by #445
